### ISSUE TITLE
ClassLoader Issue - REST API vs Server Restart

### ISSUE THREAD

**Issue by glassfishrobot** [javaee/glassfish#21695]
We have a Jenkins build that deploys a WAR via the REST API. We were not able to restart the Glassfish server when the app was already deployed (exploded in domain's applications directory). We determined that this was caused by a ClassLoader issue, so, we added the following to glassfish-web.xml.

<class-loader delegate="false"></class-loader>

Prior to making these changes I had to delete the app from the domain's applications directory, bounce the server, and redeploy from Jenkins.

The glassfish-web.xml change resolved our issue for restarting the server when the app is already deployed. However, now we are no longer able to redeploy the application from Jenkins. If I delete the application from the domain's applications directory and bounce the server I am able to deploy the WAR from Jenkins.

When I back out my changes to glassfish-web.xml I am always able to redeploy from Jenkins, but then I can't restart the server.

This is what we are using to deploy from Jenkins:
curl -k -u admin:password -X POST \
-H 'X-Requested-By: loadr' \
-H 'Accept: application/json' \
-F "id=@$

{WORKSPACE}

/erf.war" \
-F force=true https://

{host:port}

management/domain/applications/application/
#### Affected Versions
[4.1, 4.1.1]

**Comment by prsad** [comment]
This needs to be investigated since the use case could be commmon and we need to define the behaviour

**Comment by rjeberhard** [comment]
@chrisohiou57, this bug is missing the attachment of the application that failed to deploy.  Also, in the bug description is the following, "We determined that this was caused by a ClassLoader issue, so, we added the following to glassfish-web.xml."  However, there is no accompanying text that gives what was actually added.

Per management directive, we've been instructed to request missing details and to give a one-week deadline.  We will likely close this bug if the needed information is not provided.  Thanks, in advance, for your prompt response.

**Comment by chrisohiou57** [comment]
@rjeberhard Sorry for the missing information. I must have messed up when I copied the content from the original Jira ticket.

We were adding the following to glassfish-web.xml file:
<class-loader delegate="false"></class-loader>

The code has been changed since I opened this ticket, so, I won't be able to provide the WAR file. We determined that the issue was caused by our logging framework. We were deploying a Spring Boot application as a WAR file. We were using SLF4J and Logback for the logging framework. We also modified the Glassfish configuration to use SLF4J for its logging configuration.

We ended up disabling the Spring Boot logging framework using a JVM property, and went with JUL as the logging framework for the app and Glassfish. We are no longer experiencing issues with this configuration.

If you want more information about the errors we were seeing while using SLF4J and Logback I can try to reproduce it again. However, I have no issues with you closing the ticket since we have a workaround.
